Mappers/HttpMapper.php

```diff
@@ -1,226 +1,237 @@
+     * @param SearchCriteria $filter
+     * @return PaginationCollection
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws InternalException
      */
-    public function delete($identifierValue)
+    public function load(SearchCriteria $filter): PaginationCollection
     {
-        $this->sendHttpRequest('DELETE', $this->getUrlWithIdentifier($identifierValue, self::DELETE_URL), []);
-        return true;
+        $params = $this->prepareParams($filter);
+
+        $result = $this->getCollection($this->requestFactory->create(
+            self::GET,
+            $this->urlRender->prepareLoadUrl([], $params)
+        ));
+
+        $paginationCollection = new PaginationCollection(count($result), $filter->limit(), $filter->offset());
+        $paginationCollection->merge($result);
+
+        return $paginationCollection;
     }
 
     /**
-     * @param ArraySerializable $object
+     * @param array $identifiers
      * @return ArraySerializable
-     * @throws EnergonException
+     * @throws InfrastructureException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws InternalException
      */
-    public function create(ArraySerializable $object)
+    public function get(array $identifiers) : ArraySerializable
     {
-        return $this->sendHttpRequest('POST',
-            $this->getHttpMapperConfig(self::CREATE_URL),
-            ['json' => $object->toArray()]
+        return $this->getModel(
+            $this->requestFactory->create(self::GET, $this->urlRender->prepareGetUrl($identifiers))
         );
     }
 
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
+     * @param array $objectData
      * @return ArraySerializable
-     * @throws EnergonException
+     * @throws InfrastructureException
+     * @throws ValidationException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws InternalException
      */
-    public function update(ArraySerializable $object, $identifierName)
+    public function create(array $objectData): ArraySerializable
     {
-        return $this->sendHttpRequest('PUT',
-            $this->getUrlWithObjectAndIdentifier($object, $identifierName, self::UPDATE_URL),
-            ['json' => $object->toArray()]
+        return $this->getModel(
+            $this->requestFactory->create(self::POST, $this->urlRender->prepareCreateUrl($objectData), [], $objectData)
         );
     }
 
     /**
-     * @param ArraySerializable $object
-     * @param $identifierName
-     * @throws EnergonException
+     * @param array $objectData
      * @return ArraySerializable
+     * @throws InternalException
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
      */
-    public function updatePatch(ArraySerializable $object, $identifierName)
+    public function update(array $objectData)
```

Comment:
if you return type in one place please return in other, interfaces

Reply:
fixed